OCR/ocr/services/image_segmenter.py

```diff
@@ -4,8 +4,91 @@
 import os
 
 
+def crop_zeros(image):
+    # argwhere will give you the coordinates of every non-zero point
+    true_points = np.argwhere(image)
+
+    if len(true_points) == 0:
+        return None
+
+    # take the smallest points and use them as the top left of your crop
+    top_left = true_points.min(axis=0)
+    # take the largest points and use them as the bottom right of your crop
+    bottom_right = true_points.max(axis=0)
+    return image[
+        top_left[0] : bottom_right[0] + 1,  # plus 1 because slice isn't
+        top_left[1] : bottom_right[1] + 1,
+    ]  # inclusive
+
+
+def segment_by_mask_then_crop(self) -> dict[str, np.ndarray]:
+    segments = {}
+
+    with open(self.labels, "r") as f:
```

Comment:
can extract out the load function. 
def load_labels(self):
    try:
        with open(self.labels, "r") as f:
            return json.load(f)
    except (FileNotFoundError, json.JSONDecodeError) as e:
        raise ValueError(f"Error loading labels: {e}")